File: frontend/src/types/comment.ts
export type CommentTipo =
  | "comentario"
  | "cambio_estado"
  | "feedback"
  | "aprobacion"
  | "actualizacion"
  | "extension_fecha"

export interface Comment {
  id:          number
  project_id:  number
  author_id:   number
  author_role: string
  author_name: string
  message:     string
  tipo:        CommentTipo
  created_at:  string
}

File: backend/app/schemas/comment.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class CommentCreate(BaseModel):
    message: str
    tipo: Literal[
        "comentario",
        "cambio_estado",
        "feedback",
        "aprobacion",
        "actualizacion",
        "extension_fecha",
    ] = "comentario"

class CommentRead(BaseModel):
    id:          int
    project_id:  int
    author_id:   int
    author_role: str
    author_name: str
    message:     str
    tipo:        str
    created_at:  datetime

    model_config = {"from_attributes": True}
